### DISCUSSION TITLE
How to update a YAML file with scalar (|-) YAML embedded

### DISCUSSION THREAD

**Discussion by nicknarcise** [mikefarah/yq#1345] [Q&A]
I am new to YQ and trying to understand how to query and essentiallly update a YAML file that contains scalar YAML values but having a hard time understanding how to query the values and essentially update a value in place
![image](https://user-images.githubusercontent.com/44612176/190681533-ec4b9857-5e7d-4415-bb54-733ace5921fe.png)

Trying to update tag version inplace for one of the deployments but having a hard time understanding how to do this with block scalars (|-). I can seem to do "yq e '.spec.source.helm.values' but not sure how to access --> deployments for etl-one or two then update the image.tag: values

spec:
  source:
    helm:
      values: |-
        deployments:
          - name: "etl-one"
            image:
              repository: "xxxxxxxxxxxx.dkr.ecr.us-east-1.amazonaws.com/etl-one"
              tag: "v1.0.7"
          - name: "etl-two"
            image:
              repository: "xxxxxxxxxxxx.dkr.ecr.us-east-1.amazonaws.com/etl-two"
              tag: "v1.0.2"

Not sure if someone can assist me in understanding or point me to how to do this with block scalars
![block-scalar-example](https://user-images.githubusercontent.com/44612176/190682716-0b1b9bb4-6f41-4f62-bd50-db571e2c3bbd.png)

**Comment by mikefarah** [comment]
See https://mikefarah.gitbook.io/yq/operators/encode-decode#update-a-multiline-encoded-yaml-string for an example on how to update that string
